test/e2e/network/dual_stack.go

```diff
@@ -48,6 +51,9 @@ var _ = SIGDescribe("[Feature:IPv6DualStackAlphaFeature] [LinuxOnly]", func() {
 	})
 
 	ginkgo.It("should have ipv4 and ipv6 internal node ip", func() {
+		// cloud providers use the CCM to populate the node Internal IPs
+		// however the kubelet only sets one IP address as Internal IP
+		e2eskipper.SkipUnlessProviderIs("azure", "aws")
```

Comment:
Does the above mean Azure and AWS only support multiple internal IPs, not by Google cloud?

Reply:
/cc @oomichi

Reply:
I just added the ones with IPv6 support, the test title is tagged with DualStack

Reply:
this provider-key thing is a pretty gross hack we use throughout the tests

why do we have this test? should they in fact have this? if not should the test exist at all?

Reply:
less obtusely: do we actually expect all nodes to have two IPs? otherwise this test seems somewhat bogus

Reply:
@BenTheElder there is a related open issue about this https://github.com/kubernetes/kubernetes/issues/42125 , but seems that CCM populate this field.
@andrewsykim can you help us here? is it normal to use several internal ips in the cloud  providers?

Reply:
based on this discussion https://kubernetes.slack.com/archives/C718BPBQ8/p1579034801032700
we should avoid cloud providers only tests in tree.... this test will only work on cloud providers, thus each of them should implement it in their out of tree repos

Reply:
@aojea dual-stack is currently supported only in azure and now with kind. I see a lot of tests in tree still being `SkipUnlessProviderIs`. So maybe we can do the same here and once we have more providers have dual-stack support move it out of tree?

Reply:
yeah, I discussed this with the SIG-testing folks the other day and there are 2 things here.
One less important that is the general decision of the project to try to keep the e2e cloud provider agnostic and not adding more tests.
And the other, is the validity of the test as e2e, because it seems more a verification than a test, i.e. that one node has one IP of each IP family may be only a implementation detail in Azure, and basically is only testing that kubelet creates those fields, that maybe easily done with an integration test per example.
Anyway, since this created controversy and is blocking this PR I've removed the commit suppressing this e2e test and we can discuss it later.